### DISCUSSION TITLE
Custom Exporter and JavaAgent

### DISCUSSION THREAD

**Discussion by jonbranham** [open-telemetry/opentelemetry-java-instrumentation#6108] [Q&A]
I'm a bit confused about how to combine the functionality of the javaagent auto-instrumentation w/ a custom exporter created using auto-configuration and the spi hooks.   

Without the JavaAgent and using a normal classpath of dependencies for my exporter configured through Gradle, I'm able to activate the exporter through system property -Dotel.traces.exporter=abc.  The ABCSpanExporterProvider configures it using a few additional system properties that fall under the otel.traces.exporter.abc prefix.  This seems to all take place lazily upon first reference to GlobalOpenTelemetry.get().

I'd like to now combine this with auto-instrumentation as I have some messaging integrations like amqp/rabbitmq (not using Spring based APIs) and figured the javaagent would be the best fit.  When I enable the javaagent and specify it in vmargs of IntelliJ, the first issue I encountered was that it did not recognize my custom exporter named "abc".  I then tried experimenting with otel.javaagent.extensions=/abc-exporter.jar and this got a little bit further, however some of the dependencies that this exporter jar used we unable to be loaded.

Does this custom exporter need to be packaged in a certain way with all transitive dependencies?  Or is a directory supported in which the exporter and transitive dependencies can be placed?

**Comment by trask** [comment]
hi @jonbranham! check out https://github.com/open-telemetry/opentelemetry-java-instrumentation/tree/main/examples/extension

**Reply by jonbranham** [reply]
Interesting so the gradle build is using a custom configuration to collect otel and extension deps and the shadow plugin to combine and emit a jar.  I see a comment in there about most of this boilerplate not being necessary with an otel plugin.  Is there any info on this plugin or an eta?

**Reply by laurit** [reply]
If you mean the TODO then it should refer to our `io.opentelemetry.instrumentation.javaagent-instrumentation` gradle plugin https://github.com/open-telemetry/opentelemetry-java-instrumentation/blob/main/conventions/src/main/kotlin/io.opentelemetry.instrumentation.javaagent-instrumentation.gradle.kts which could allow removing the 5 lines following the TODO

**Comment by jkwatson** [comment]
Moving this to the instrumentation project..
